### ISSUE TITLE
Documentation for trying the development version of Astropy ignores virtual environments

### ISSUE THREAD

**Issue by janerigby** [astropy/astropy#11207]
### Description
I am trying to contribute to AstroPy.  The tutorial, "Try the Development Version" 
https://docs.astropy.org/en/stable/development/workflow/get_devel_version.html#get-devel
shows a method to switch over to the development version, but warns against doing so, and says that using python virtual environments is a much better choice. It then links to a general article about virtual environments for astropy, https://docs.astropy.org/en/stable/development/workflow/virtual_pythons.html#virtual-envs, that doesn't help with the specific case of making a virtual environment for astropy development testing.  

It would be much better if the "Try the development version" tutorial just showed how to do so using virtual environments, since that's the recommended method.

**Comment by pllim** [comment]
Indeed, the dev docs need updating. Everyone has their own favorite envs, so it is tough to cover all the basis. Perhaps @embray can advise here.

In addition, I would recommend replacing "stable" with "latest" in the URLs above to ensure you obtain the latest instructions.

In the meantime, if you need help, feel free to reach out to us on [Astropy Slack](http://joinslack.astropy.org/).

**Comment by janerigby** [comment]
From @taldcroft on Slack:
`conda create -n astropy-dev python=3.7 
conda activate astropy-dev
cd ~/git/astropy  # my git clone
pip install pytest-astropy
pip install -e '.[test,all,docs]'  # Quotes needed for zsh`

**Comment by mhvk** [comment]
As @pllim noted, part of the problem is that there are so many options used even between ourselves. As most testing is done with `tox`, I've slowly been moving towards that and even have #10394 with a description for setup using that...

**Comment by embray** [comment]
I agree though that for development you should absolutely use one of

* venv
* virtualenv
* conda

This should be clarified.  If there are other options, people already using those options can be presumed to know what they're doing so it doesn't need to be an exhaustive list.  venv comes with Python 3 so it is the easiest, and should be by default part of the instructions, with mentions also of virtualenv and conda for those already using them.
